src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -118,38 +121,60 @@ public void removeModelWaitingForAssessment(long exerciseId, long modelSubmissio
     }
 
     /**
+     * Get the (cached) list of models that need to be assessed next. If the number of models in the list is smaller than the configured NUMBER_OF_OPTIMAL_MODELS a new "optimal"
+     * model is added to the list. The models in the list are optimal in the sense of knowledge gain for Compass, helping to automatically assess as many other models as possible.
+     *
      * @param exerciseId the exerciseId
      * @return List of model Ids waiting for an assessment by an assessor
      */
-    public Set<Long> getModelsWaitingForAssessment(long exerciseId) {
+    public List<Long> getModelsWaitingForAssessment(long exerciseId) {
```

Comment:
Even though it doesn't change something logic wise, I think it was better when it was a Set (just for readability).

Reply:
You are right. The reason I changed it to a `List` is that I shuffle the returned list in the `getModelingSubmissionWithoutResult` method of the `ModelingSubmissionService`. As a `Set` is unordered and cannot be shuffled I would have to transform the `Set` after calling this method to be able to shuffle it. I had it like this in a previous version and I thought it increases readability of the calling method if we just return a `List` here. 

The reason for shuffling is that we want to prevent that the user gets the same submission over and over again after canceling its assessment.

Reply:
Do you think it would be possible keep it a set in the method, and then convert it to a list when it needs shuffling?
`List<Long> list = new ArrayList(set);`

I don't think it's a big problem though, so don't bother with it if you feel it's ok.

Reply:
As discussed not a good idea as the list was previously sorted.